src/plugin/timezone/index.js

```diff
@@ -97,8 +97,13 @@ export default (o, c, d) => {
     const date = this.toDate()
     const target = date.toLocaleString('en-US', { timeZone: timezone })
     const diff = Math.round((date - new Date(target)) / 1000 / 60)
-    let ins = d(target).$set(MS, this.$ms)
-      .utcOffset((-Math.round(date.getTimezoneOffset() / 15) * 15) - diff, true)
+    const offset = -date.getTimezoneOffset() - diff
+    if (offset === 0) {
+      return this.utc(keepLocalTime)
```

Comment:
First of all thanks for this changes. We were struggling with this issues too.

There is small issue with this line. If we call this with some specific timezone like Affrica/Accara which is also timezone with offset 0, timezone information is not stored on dayjs object. With following code that information will be stored as well in this case too
```
  const x = this.utc(keepLocalTime)
  x.$x.$timezone = timezone 
  return x
```

This is useful later on if we want to find out exactly what timezone is used on an object. 
For example moment before had option to call date.tz() to get timezone out of it. We internally created small plugin to get the same from dayjs object.

This feature is also requested from a lot of people here so it would be nice to add it maybe as part of this PR if this will be ever merged.

```
 d.timezone = function(){
      return this.$x.$timezone
 }
```

Reply:
I added some tests to show 'offsetName' and the timezone name. These tests show that the timezone name is preserved even for timezones on the GMT meridian. But dayjs.tz uses the timezone returne by INTL e.g. for formatting (that's what you get when using `.offsetName('short')`).

So a function that returns the 'true' timezone name really would be fine; but that a case for another PR.
As a small improvement to your proposal, I would add 'guess()' to make sure that you always get a timezone name.
```
  proto.timezoneName = function () {
    return this.$x.$timezone || d.tz.guess()
  }
```

So feel free to make another PR; perhaps such a small one gets accepted faster 😄